### ISSUE TITLE
10 minute delay before sending "Reply contact is..."

### ISSUE THREAD

**Issue by GoogleCodeExporter** [baluragala/gtalksms#266]
```
Relevant part of the log is:



04-16 15:55:55.171 I/gtalksms( 7840): XMPP packet received - sending Intent: 
com.googlecode.gtalksms.action.XMPP.MESSAGE_RECEIVED
04-16 15:55:55.171 I/gtalksms( 7840): handling action 
'com.googlecode.gtalksms.action.XMPP.MESSAGE_RECEIVED' while in state Connected
04-16 15:55:55.171 D/gtalksms( 7840): MainService executing command: 
"sms:#REDACTED#:REDACTED"
04-16 15:55:59.631 I/gtalksms( 7840): handled action 
'com.googlecode.gtalksms.action.XMPP.MESSAGE_RECEIVED' - state now: Connected
04-16 16:06:16.861 I/gtalksms( 7840): Sending message "Reply contact is 
REDACTED"
04-16 16:06:30.881 I/gtalksms( 7840): onStartCommand(): Intent 
com.googlecode.gtalksms.action.CONNECT
04-16 16:06:30.881 I/gtalksms( 7840): handling action 
'com.googlecode.gtalksms.action.CONNECT' while in state Connected
04-16 16:06:30.881 I/gtalksms( 7840): handled action 
'com.googlecode.gtalksms.action.CONNECT' - state now: Connected
04-16 16:06:31.071 I/gtalksms( 7840): Update presence: REDACTED
04-16 16:06:31.071 I/gtalksms( 7840): Update presence: REDACTED


Why the ten minutes delay?
```

Original issue reported on code.google.com by `antoine....@gmail.com` on 16 Apr 2012 at 2:16
- Blocked on: #268

**Comment by GoogleCodeExporter** [comment]
```
Thanks for the log.

The reply contact notification get's send delayed by 10 seconds via an extra 
thread. So OS scheduling could add some more time, but it should be only within 
the seconds region. Also this thread issues just an send Intent, which has to 
go down into Android's Intent handling mechanism and then injected back to 
GTalkSMS. This also could add an extra delay.

Anyway, we should add an extra log information when the thread issues his 
intent and on some parts of the code.

Do you experience the delay every time or was this a one time issue? I also 
have sometimes delays up to a few minutes but in 95% the "reply contact is" 
comes within the expected 20 seconds.
```

Original comment by `fschm...@gmail.com` on 17 Apr 2012 at 12:11
- Changed title: **10 minute delay before sending "Reply contact is..."**
- Changed state: **Accepted**
- Added labels: **Priority-Low**
- Removed labels: **Priority-Medium**

**Comment by GoogleCodeExporter** [comment]
```
It depends. I just got a three-minute delay. Usually it's fine though.
```

Original comment by `antoine....@gmail.com` on 17 Apr 2012 at 12:17

**Comment by GoogleCodeExporter** [comment]
```
Hi,

Anything I can do to help debug this? It's really annoying, and causes me to 
send messages to the wrong people.
```

Original comment by `antoine....@gmail.com` on 26 Apr 2012 at 7:14

**Comment by GoogleCodeExporter** [comment]
```
Hi,

Can you provide us more log (with system log if possible)?
Which phone do you have? Which Android version?

The workaround for you is to use the command sms:#contact# instead of the reply 
command until we fix it.

Thanks
```

Original comment by `Florent....@gmail.com` on 30 Apr 2012 at 8:07

**Comment by GoogleCodeExporter** [comment]
```
Hi,

The log is always the same than the one I copied. I'll get the system log and 
post it here next time it happens. I have a HTC nexus one, android 2.3.4 (for 
some reason I can't do the upgrade to 2.3.6). 

Yes, that's what I use now, but it's tiresome (plus, I've hacked my jabber 
client to automatically add reply: in front of the messages I send). I'd be 
much happier if reply worked reliably. Also, could you add a setting for the 
delay ? I think the default is ten seconds, I'd like to set it to five or 
something like that.
```

Original comment by `antoine....@gmail.com` on 30 Apr 2012 at 10:06

**Comment by GoogleCodeExporter** [comment]
```
Unfortunately I think this is by design. We will add more debug output for the 
reply command in the next release or beta of GTalkSMS. But this will only show 
the cause of the delay. A fix will likely also need a new design of the 
recipient notification.

Please open a new enhancement request in the issue tracker for the setting of 
the delay time.
```

Original comment by `fschm...@gmail.com` on 30 Apr 2012 at 1:54

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision aa87994a59bb.
```

Original comment by `fschm...@gmail.com` on 26 Jan 2013 at 5:56
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
New version 4.4 Beta 1 is available in the download section
```

Original comment by `Florent....@gmail.com` on 16 Feb 2013 at 9:47

**Comment by GoogleCodeExporter** [comment]
```
Fixed in 4.4
```

Original comment by `Florent....@gmail.com` on 15 Sep 2013 at 3:27
- Changed state: **Deployed**
